gsy_framework/schema/validators.py

```diff
@@ -49,6 +56,128 @@ def validate(self, data, raise_exception: bool = False) -> (bool, str):
             return False, str(exc)
 
 
+class AVROSchemaSerializer(AVROSchemaValidator):
+    """Serializer class that uses AVRO schemas for serializing and deserializing dicts to bytes."""
+    # pylint: disable=no-self-use
+
+    def modify_input(self, data: Dict) -> Dict:
```

Comment:
I find the `modify_input`, `modify_output`, `serialize` and `deserialize` methods very useful. However, I suggest to also add them to BaseSchemaValidator as abstract methods to abstract implementation details from the using code.

Reply:
I was also in-between this actually, and I did not do this in order to distinguish pure validators from validators + serializers. On second thought though you are right, I think that we can have only serializer classes that expose the validate method separately, that way the same class can be used to just validate without raising exceptions and also to serialize / deserialize depending on the users' desire. Fixing this